src/Components/Endpoints/src/DependencyInjection/HttpNavigationManager.cs

```diff
@@ -11,7 +11,7 @@ internal sealed class HttpNavigationManager : NavigationManager, IHostEnvironmen
 
     protected override void NavigateToCore(string uri, NavigationOptions options)
     {
-        var absoluteUriString = ToAbsoluteUri(uri).ToString();
+        var absoluteUriString = ToAbsoluteUri(uri).AbsoluteUri;
```

Comment:
Accessing the `Uri.AbsoluteUri` property will throw if the `Uri` instance does not represent an absolute URI, but this case is safe because `ToAbsoluteUri()` always returns an absolute URI.